### PR TITLE
a better description of the build system script

### PR DESCRIPTION
**Pull request by g4titanx**

### CODE REVIEW

**Comment by g4titanx** [comment]
i dont know why, but some of the tests are failing

File: build_system/src/main.rs [hunk 1]
```diff
@@ -26,17 +26,29 @@ macro_rules! arg_error {
...
+
+        cargo     : Executes a cargo command. Use 'cargo --help' for a list of cargo commands.
+        clean     : Cleans the build directory, removing all compiled files and artifacts.
+        prepare   : Prepares the environment for building, including fetching dependencies and setting up configurations.
+        build     : Compiles the project. Use 'build --help' for build options.
+        test      : Runs tests for the project. Use 'test --help' for test options.
+        info      : Displays information about the build environment and project configuration.
+        clone-gcc : Clones the GCC compiler from a specified source. Use 'clone-gcc --help' for options.
+
+Examples:
```
Comment (GuillaumeGomez) [inline build_system/src/main.rs:46]: I don't think having examples in a good idea here. Better let people check the `Readme.md` file instead.

File: build_system/src/build.rs [hunk 2]
```diff
@@ -26,9 +27,7 @@ impl BuildArg {
                         build_arg.flags.push("--features".to_string());
                         build_arg.flags.push(arg.as_str().into());
                     } else {
-                        return Err(
-                            "Expected a value after `--features`, found nothing".to_string()
-                        );
+                        return Err("Expected a value after `--features`, found nothing".to_string());
```
Comment (GuillaumeGomez) [inline build_system/src/build.rs:30]: CI is failing because it's not happy with this change. (Wrong formatting)

**Comment by GuillaumeGomez** [comment]
Also, why are there 10 commits?

File: build_system/src/main.rs (hunk 1 above)
Comment (g4titanx) [reply build_system/src/main.rs:46]: oh, alright

**Comment by g4titanx** [comment]
> Also, why are there 10 commits?

there were supposed to be just two, I think it's between when I made a git pull, committed my changes, and then made a git push. it accounted for the recent commits
I don't know why it did that tho

File: build_system/src/build.rs (hunk 2 above)
Comment (g4titanx) [reply build_system/src/build.rs:30]: reverted the change
i am gonna commit now and see what happens

File: build_system/src/main.rs [hunk 3]
```diff
@@ -26,17 +26,29 @@ macro_rules! arg_error {
...
-    prepare   : Run prepare command
-    build     : Run build command
-    test      : Run test command
-    info      : Run info command
-    clone-gcc : Run clone-gcc command
-    --help    : Show this message"
+Usage: build_system [command] [options]
+
+Options:
+        -h, --help    : Displays this help message.
```
[the remark this answers is not in the document]
Comment (g4titanx) [reply build_system/src/main.rs:34]: okay

**Comment by GuillaumeGomez** [comment]
You still have 11 commits.

File: build_system/src/main.rs [hunk 4]
```diff
@@ -26,17 +26,23 @@ macro_rules! arg_error {
...
+Options:
+        --help    : Displays this help message.
+
+Commands:
+
+        cargo     : Executes a cargo command. Use 'cargo --help' for a list of cargo commands.
+        clean     : Cleans the build directory, removing all compiled files and artifacts.
+        prepare   : Prepares the environment for building, including fetching dependencies and setting up configurations.
+        build     : Compiles the project. Use 'build --help' for build options.
+        test      : Runs tests for the project. Use 'test --help' for test options.
```
Comment (antoyo) [inline build_system/src/main.rs:42]: It is redundant to mention `--help` for every command. Please remove it.

**Comment by g4titanx** [comment]
> You still have 11 commits.

well, that's because I was working on that issue with a different GitHub account that got flagged and I had to create a new one. so I committed the changes I had made

**Comment by g4titanx** [comment]
hello @GuillaumeGomez , i have being trying to squash my commits and I haven't being to do that, this is my first attempt at trying to squash a commit tho. i have been at it since you mentioned and it turns rebase squash wasn't as easy as I thought.

**Comment by GuillaumeGomez** [comment]
[A tutorial on how to squash commits](https://blog.guillaume-gomez.fr/Git-tips/1/2).

**Comment by g4titanx** [comment]
tha

> [A tutorial on how to squash commits](https://blog.guillaume-gomez.fr/Git-tips/1/2).

thank you!

**Comment by GuillaumeGomez** [comment]
You only keep on adding new commits...

**Comment by g4titanx** [comment]
> You only keep on adding new commits...

yeah, i realized that, i am trying to figure it out

**Pull request closed without merging**